test/s3-benchmarks/src/main/java/software/amazon/awssdk/s3benchmarks/BaseJavaS3ClientBenchmark.java

```diff
@@ -0,0 +1,130 @@
+    private final ChecksumAlgorithm checksumAlgorithm;
+    private final int iteration;
+
+    protected BaseJavaS3ClientBenchmark(TransferManagerBenchmarkConfig config) {
+        this.bucket = Validate.paramNotNull(config.bucket(), "bucket");
+        this.key = Validate.paramNotNull(config.key(), "key");
+        this.timeout = Validate.getOrDefault(config.timeout(), () -> DEFAULT_TIMEOUT);
+        this.iteration = Validate.getOrDefault(config.iteration(), () -> BENCHMARK_ITERATIONS);
+        this.checksumAlgorithm = config.checksumAlgorithm();
+
+        this.s3Client = S3Client.create();
+
+        long partSizeInMb = Validate.paramNotNull(config.partSizeInMb(), "partSize");
+        long readBufferInMb = Validate.paramNotNull(config.readBufferSizeInMb(), "readBufferSizeInMb");
+        Validate.mutuallyExclusive("cannot use forceCrtHttpClient and connectionAcquisitionTimeoutInSec",
+                                   config.forceCrtHttpClient(), config.connectionAcquisitionTimeoutInSec());
+        this.s3AsyncClient = S3AsyncClient.builder()
+                                          .multipartEnabled(true)
+                                          .multipartConfiguration(c -> c.minimumPartSizeInBytes(partSizeInMb * MB)
+                                                                        .thresholdInBytes(partSizeInMb * 2 * MB)
+                                                                        .apiCallBufferSizeInBytes(readBufferInMb * MB))
+                                          .httpClient(httpClient(config))
+                                          .build();
+    }
+
+    private SdkAsyncHttpClient httpClient(TransferManagerBenchmarkConfig config) {
+        if (config.forceCrtHttpClient()) {
+            logger.info(() -> "Using CRT HTTP client");
+            AwsCrtAsyncHttpClient.Builder builder = AwsCrtAsyncHttpClient.builder();
+            if (config.readBufferSizeInMb() != null) {
+                builder.readBufferSizeInBytes(config.readBufferSizeInMb());
+            }
+            if (config.maxConcurrency() != null) {
+                builder.maxConcurrency(config.maxConcurrency());
+            }
+            return builder.build();
+        }
+        NettyNioAsyncHttpClient.Builder builder = NettyNioAsyncHttpClient.builder();
+        if (config.connectionAcquisitionTimeoutInSec() != null) {
+            Duration connAcqTimeout = Duration.ofSeconds(config.connectionAcquisitionTimeoutInSec());
+            builder.connectionAcquisitionTimeout(connAcqTimeout);
+        }
+        if (config.maxConcurrency() != null) {
+            builder.maxConcurrency(config.maxConcurrency());
+        }
+        return builder.build();
+    }
+
+    protected abstract void sendOneRequest(List<Double> latencies) throws Exception;
+
+    protected abstract long contentLength() throws Exception;
+
+    @Override
+    public void run() {
+        try {
+            warmUp();
+            doRunBenchmark();
```

Comment:
Dongie already proposed it, yeah! Indeed, it would be nice to plan some time to move this package to using JMH in the future. For now that was not a priority.